app/code/Magento/CatalogGraphQl/Model/AttributesJoiner.php

```diff
@@ -124,4 +170,12 @@ private function setSelectionsForFieldNode(FieldNode $fieldNode, array $selected
     {
         $this->queryFields[$fieldNode->name->value][$fieldNode->name->loc->start] = $selectedFields;
     }
+
+    /**
+     * @param ResolveInfo $resolverInfo
+     */
+    public function setResolverInfo(ResolveInfo $resolverInfo): void
```

Comment:
I don't think we should set the resovlerInfo in this way. For one it introduces temporal coupling.
Can we pass down resolve info via methods arguments instead.

Reply:
removed class member and used method argument